src/rpc/rawtransaction.cpp

```diff
@@ -779,7 +782,6 @@ UniValue decodescript(const UniValue& params, bool fHelp)
             + HelpExampleRpc("decodescript", "\"hexstring\"")
         );
```

Comment:
The `cs_main` lock on `decoderawtransaction` could probably also have been removed, I think. Does not block.

Reply:
I don't think so, because `decoderawtransaction` calls `TxToJSON()` which calls a couple of the `chainActive` methods. Upstream `TxToJSON()` internally locks `cs_main`, which we probably should do too. (This may appear in a later PR in this series.)